chrome/views/popupView.js

```diff
@@ -63,68 +66,128 @@ GmailToTrello.PopupView.prototype.confirmPopup = function() {
+    // If we need to load the cards for this thread
+    if (!self.parent.model.isThreadCardsLoaded) {
+        $button.detach();
+        $popup.detach();
+        gtt_log('PopupView:confirmPopup: getting data for button');
+        self.parent.model.loadThreadTrelloCards(); // loads cards, creates html, then circles back to confirmPopup
+        return;
+    }
+
+    // If the button already exists on the page
+    if ($button.length > 0) {
+        if ($button.first().is(":visible")) {
+            gtt_log('PopupView:confirmPopup: button visible');
         } else {
-            var img = 'GtT';
-            
-            // Refresh icon present? If so, use graphics, if not, use text:
-            if ($('div.asl.T-I-J3.J-J5-Ji,div.asf.T-I-J3.J-J5-Ji', this.$toolBar).length > 0) {
-                img = '<img class="f tk3N6e-I-J3" height="13" width="13" src="'
-                  + chrome.extension.getURL('images/icon-13.jpg')
-                  + '" />';
+            gtt_log('PopupView:confirmPopup: Button is in an inactive region. Moving...');
+            //relocate
+            if ($button.length > 1) {
+                $button.detach(); // In case multiple copies were created
+                if ($popup.length > 1) {
+                    $popup.detach(); // In case copies were created
+                }
             }
-
-            this.html['add_to_trello'] =
-                '<div id="gttButton" class="T-I J-J5-Ji ar7 nf T-I-ax7 L3" ' // "lS T-I-ax7 ar7"
-                  + 'data-tooltip="Add this Gmail to Trello">'
-                  + '<div aria-haspopup="true" role="button" class="J-J5-Ji W6eDmd L3 J-J5-Ji Bq L3" tabindex="0">' // class="J-J5-Ji W6eDmd L3 J-J5-Ji Bq L3">' // 
-                  + img
-                  + '<div id="gttDownArrow" class="G-asx T-I-J3 J-J5-Ji">&nbsp;</div></div></div>';
+            gtt_log('PopupView:confirmPopup: adding Button and Popup');
+            $button.first().appendTo(this.$toolBar);
+            $popup.first().appendTo(this.$toolBar);
+        }
+        if ($popup.length < 1) {
+            self.addOrCreatePopup();
         }
+        return;
+    }
+
+    // If the button doesn't exist but the html is set for it
+    if (this.html && this.html['add_to_trello'] && this.html['add_to_trello'].length > 0) {
         gtt_log('PopupView:confirmPopup: creating button');
         this.$toolBar.append(this.html['add_to_trello']);
-        needInit = true;
-    } else if ($button.first().is(":visible")) {
-        gtt_log('PopupView:confirmPopup: button visible');
-    } else {
-        gtt_log('PopupView:confirmPopup: Button is in an inactive region. Moving...');
-        //relocate
-        if ($button.length > 1) {
-            $button.detach(); // In case multiple copies were created
-            if ($popup.length > 1) {
-                $popup.detach(); // In case copies were created
+        self.addOrCreatePopup();
+        return;
+    }
+
+};
```

Comment:
I rewrote this function a bit and pulled out the button HTML generation into another function. There are three cases it now handles. Would probably be better to be separate functions, really...

1. When it's first called, the thread cards aren't loaded, so it needs to load the thread cards. This function calls loadThreadCards, which calls buildPopupHtml, which calls this function again.
2. Now that the thread cards are loaded, and this.html is set, this function runs addOrCreatePopup to append the button.
3. If there is already a button in place when this is called, it just run the logic that was here before.